### PR TITLE
DSE 5.1.41 release notes

### PR DESCRIPTION
**Pull request by szymon-miezal**
----
## Release Notes Automation
If you name your pull-request as "Product x.y.z Release ...", after merging the
PR, a GitHub Action will automatically create a product version tag "product-x.y.z".

Supported product names are:
 * DSE
 * OpsCenter
 * Studio
 * Luna Streaming

Version supports 3 sets or 4 sets of digits.

### CODE REVIEW

File: DSE_5.1_Release_Notes.md [hunk 1]
```diff
@@ -1,6 +1,25 @@
 # Release notes for DataStax Enterprise 5.1
 DSE 5.1.x is compatible with Apache Cassandra&trade; 3.11 and adds additional production-certified changes, if any. Components that are indicated with an asterisk (&ast;) (if any) are known to be updated since the prior patch version.
 
+# Release notes for 5.1.41
+12 October 2023
```
Comment (thobe) [inline DSE_5.1_Release_Notes.md:5]: This date is wrong!

**Review by thobe — CHANGES_REQUESTED** [review]
The release date should be November 6 (or something like that), not October 12.

File: DSE_5.1_Release_Notes.md (hunk 1 above)
Comment (szymon-miezal) [reply DSE_5.1_Release_Notes.md]: Thank you for noticing it! As we found out, the dates are taken from jira releases which are generated by the release job, it turns out that logic does not generate dates according to our expectations.

**Review by thobe — APPROVED** [review]
This looks good now.

**Pull request merged**